core/jvm/src/main/scala/scalaz/zio/RTS.scala

```diff
@@ -633,7 +628,7 @@ private object RTS {
               // Interruption cannot be interrupted:
               this.noInterrupt += 1
 
-              curIo = IO.terminate[E, Any](die.get)
+              curIo = IO.terminate[E, Any](die.getOrElse(Nil): _*)
```

Comment:
It should probably be left as `die.get` because if it's empty, there's a bug in the code and we'd like to know about it. The only reason `fold` et al aren't used is the extra allocation overhead.